zk-keygen/src/main.rs

```diff
@@ -0,0 +1,371 @@
+                .arg(
+                    Arg::new("outfile")
+                        .short('o')
+                        .long("outfile")
+                        .value_name("FILEPATH")
+                        .takes_value(true)
+                        .help("Path to generated file"),
+                )
+                .arg(
+                    Arg::new("force")
+                        .short('f')
+                        .long("force")
+                        .help("Overwrite the output file if it exists"),
+                )
+                .arg(
+                    Arg::new("silent")
+                        .short('s')
+                        .long("silent")
+                        .help("Do not display seed phrase. Useful when piping output to other programs that prompt for user input, like gpg"),
+                )
+                .key_generation_common_args()
+                .arg(no_outfile_arg().conflicts_with_all(&["outfile", "silent"]))
+        )
+}
+
+fn main() -> Result<(), Box<dyn error::Error>> {
+    let matches = app(solana_version::version!())
+        .try_get_matches()
+        .unwrap_or_else(|e| e.exit());
+    do_main(&matches).map_err(|err| DisplayError::new_as_boxed(err).into())
+}
+
+fn do_main(matches: &ArgMatches) -> Result<(), Box<dyn error::Error>> {
+    let subcommand = matches.subcommand().unwrap();
+    match subcommand {
+        ("new", matches) => {
+            let key_type = match matches.value_of("key_type").unwrap() {
+                "elgamal" => KeyType::ElGamal,
+                "symmetric" => KeyType::Symmetric,
+                _ => unreachable!(),
+            };
+
+            let mut path = dirs_next::home_dir().expect("home directory");
+            let outfile = if matches.is_present("outfile") {
+                matches.value_of("outfile")
+            } else if matches.is_present(NO_OUTFILE_ARG.name) {
+                None
+            } else {
+                path.extend([".config", "solana", &key_type.default_file_name()]);
+                Some(path.to_str().unwrap())
+            };
+
+            match outfile {
+                Some(STDOUT_OUTFILE_TOKEN) => (),
+                Some(outfile) => check_for_overwrite(outfile, matches)?,
+                None => (),
+            }
+
+            let word_count: usize = matches.value_of_t(WORD_COUNT_ARG.name).unwrap();
+            let mnemonic_type = MnemonicType::for_word_count(word_count)?;
+            let language = acquire_language(matches);
+
+            let mnemonic = Mnemonic::new(mnemonic_type, language);
+            let (passphrase, passphrase_message) = acquire_passphrase_and_message(matches).unwrap();
+            let seed = Seed::new(&mnemonic, &passphrase);
+
+            match key_type {
+                KeyType::ElGamal => {
+                    let silent = matches.is_present("silent");
+                    if !silent {
+                        println!("Generating a new ElGamal keypair");
```

Comment:
```suggestion
                        eprintln!("Generating a new ElGamal keypair");
```

consider putting "status update" text out on stderr instead of stdout

Reply:
Done!